### PR TITLE
Add helper to run CIs for a branch using `az pipelines`.

### PR DESCRIPTION
**Pull request by skottmckay**
### Description

Add helper to run CIs for a branch using `az pipelines`.
This can be used to easily kick off multiple CIs for a branch prior to creating a PR.

Update run_CIs_for_external_pr.py so the CI list can be shared. 
Request json output from `gh pr view` so the current state is more easily parsed.

### CODE REVIEW

File: tools/python/run_CIs_for_branch.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        description="""Run the CIs used to validate PRs for the specified branch.
+
+        Requires the Azure CLI with DevOps extension to be installed.
+          Azure CLI: https://learn.microsoft.com/en-us/cli/azure/install-azure-cli
+          DevOps extension: https://github.com/Azure/azure-devops-cli-extension
+
+        Configuration:
+          Login:`az login`
+          Configure ORT repo as default:
+            `az devops configure --defaults organization=https://dev.azure.com/onnxruntime project=onnxruntime`
```
Comment (xadupre) [inline tools/python/run_CIs_for_branch.py:29]: Should we add an example of the command line to run to use that command?

Comment (skottmckay) [reply tools/python/run_CIs_for_branch.py:29]: Sorry - which command do you mean? The `az` commands to configure or for the actual script? 

Latter could be something like 
`python .\tools\python\run_CIs_for_branch.py --exclude training --include Linux me/MyBranch`

**Comment by fs-eire** [comment]
Is the motivation of this change is because "/azp run" not working?

**Comment by skottmckay** [comment]
> Is the motivation of this change is because "/azp run" not working?

It was initially, but when I tested it the PR didn't seem to pickup that the CI had run. I have in the past wanted to kick off a few CIs when developing and this provides a simple way to do that prior to creating a PR.

File: tools/python/run_CIs_for_branch.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            `python run_CIs_for_branch.py --include linux my/BranchName`
+          Exclude training CIs
+            `python run_CIs_for_branch.py --exclude training my/BranchName`
+          Run non-training Linux CIs
+            `python run_CIs_for_branch.py --include linux --exclude training my/BranchName`
+        """,
+    )
+
+    parser.add_argument("-i", "--include", type=str, help="Include CIs that match this string. Case insensitive.")
+    parser.add_argument("-e", "--exclude", type=str, help="Exclude CIs that match this string. Case insensitive.")
```
Comment (edgchen1) [inline tools/python/run_CIs_for_branch.py:46]: maybe mention that the exclude pattern is processed after any include pattern.

**Review by edgchen1 — APPROVED** [review]

**Review by edgchen1 — DISMISSED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**